dags/mag_and_friends.py

```diff
@@ -0,0 +1,491 @@
+        'study': study, 
+        'timestamp': timestamp, 
+        'mag_revision': mag_revision,
+        'metatdenovo_revision': metatdenovo_revision,
+        'taxprofiler_revision': taxprofiler_revision
+    }
+    with open(provenance_path, 'a', newline='') as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        if os.stat(provenance_path).st_size == 0:
+            writer.writeheader()
+        writer.writerow(new_data)
+
+def create_dag():
+    default_args = {
+        'start_date': pendulum.datetime(2021, 1, 1, tz="UTC")
+    }
+
+    with DAG(
+        dag_id="automated_mag_and_friends",
+        schedule_interval=None,
+        default_args=default_args,
+        catchup=False,
+        params={'clusterLogin': 'dcallan'}
+    ) as dag:
+        # this dag will run jobs on pmacs
+        cluster_config = json.load(open(CLUSTER_CONFIG_PATH))
+        cluster_manager = ClusterManager(
+            cluster_config['headNode'], 
+            cluster_config['fileTransferNode'], 
+            cluster_config['clusterType'], 
+            "{{params.clusterLogin}}"
+        )
+
+        copy_pipeline_configs = TaskGroup("copy_pipeline_configs")
+        manage_reference_dbs = TaskGroup("manage_reference_dbs")
+
+        # TODO do we still want to do this if there are no studies to process?
+        copy_mag_config_to_cluster = cluster_manager.copyToCluster(
+            BASE_PATH, 
+            'mag.config', 
+            '.', 
+            gzip=False, 
+            task_id = "copy_mag_config_to_cluster",
+            task_group = copy_pipeline_configs
+        )
+
+        copy_fetchngs_config_to_cluster = cluster_manager.copyToCluster(
+            '/data/MicrobiomeDB/common',
+            'fetchngs.config',
+            '.',
+            gzip=False,
+            task_id = "copy_fetchngs_config_to_cluster",
+            task_group = copy_pipeline_configs
+        )
+
+        copy_taxprofiler_config_to_cluster = cluster_manager.copyToCluster(
+            BASE_PATH,
+            'taxprofiler.config',
+            '.',
+            gzip=False,
+            task_id = "copy_taxprofiler_config_to_cluster",
+            task_group = copy_pipeline_configs
+        )
+
+        copy_metatdenovo_config_to_cluster = cluster_manager.copyToCluster(
+            BASE_PATH,
+            'metatdenovo.config',
+            '.',
+            gzip=False,
+            task_id = "copy_metatdenovo_config_to_cluster",
+            task_group = copy_pipeline_configs
+        )
+
+        copy_samplesheet_templates_to_cluster = cluster_manager.copyToCluster(
+            BASE_PATH,
+            'samplesheet_templates',
+            '.',
+            gzip=False,
+            task_id = "copy_samplesheet_templates_to_cluster"
+        )
+
+        get_kraken_db = cluster_manager.startClusterJob(
+            "if [[ ! -f k2_pluspf_20240112.tar.gz ]]; then wget https://genome-idx.s3.amazonaws.com/kraken/k2_pluspf_20240112.tar.gz; fi;",
```

Comment:
Would it be better to have the db's url as a variable? In case something changes on the DB side? Just curious about when to set hard coded path/url and when to use variables.

Reply:
i could go either way on this. i dont think the url is something well potentially want to reference multiple times in the dag, or necessarily want to change very frequently. so in my mind, id be ok w it as a var but dont see it as urgent